patches/net/minecraft/client/sounds/MusicManager.java.patch

```diff
@@ -0,0 +1,14 @@
+--- a/net/minecraft/client/sounds/MusicManager.java
++++ b/net/minecraft/client/sounds/MusicManager.java
+@@ -24,7 +_,10 @@
+     }
+ 
+     public void tick() {
+-        Music music = this.minecraft.getSituationalMusic();
++        Music music = net.neoforged.neoforge.client.ClientHooks.selectMusic(this.minecraft.getSituationalMusic());
++        if (music == null) {
```

Comment:
What is null supposed to do here? Is this stopping the music or keeping the current music playing?

Reply:
yeah, if the music is set to null by the event, it’s meant to cancel the music from potentially playing

Reply:
the music only should be *stopped and replaced* if the Music object specifies it, as in if `Music#replaceCurrentMusic` is true
If the music is set to null, it will not try to play the next music track
Otherwise, and if the music is *not* meant to replace the current music, it will use that music track for the cooldown logic (such as choosing how long until the next music track should play), and if the already-playing music has stopped and the cooldown has finished, the music set by the event is the one that plays.

Reply:
Ok, what does 'cancel the music from potentially playing mean'? Does that mean stop music altogether or stop any other modifications to the song from being played, because both have their issues.

Returning `null` just keeps the current song playing until it eventually stops. Then, no song will be played **at all** until some song is returned and the `nextSongDelay` finishes its countdown, which could be a long realm of silence. The quick end of the logic seems to have weird behavior.

I think there should be a way to stop all music, and that the `nextSongDelay` isn't blocked from the music being stopped.